### DISCUSSION TITLE
1주차 사전 과제, 이예린

### DISCUSSION THREAD

**Discussion by yeyerinrin** [MrMirror21/2024-KHU-CSE-TrackStudy-Web#22] [General]
- [ ]  명명 규칙(Naming Convention)

1. Camel Case
- 각 단어의 첫 문자를 대문자로 표기하되, 이름의 첫 문자는 소문자로 작성.
2. Pascal Case
- 각 단어의 첫 문자를 대문자로 표기하며, 이름의 첫 문자 또한 대문자로 작성.
3. Snake Case
- 각 단어 사이를 언더바(_)로 구분.
4. Kebab Case
- 각 단어 사이를 하이픈(-)으로 구분.
5. Hungarian Notation
- 이름 앞에 변수의 타입을 접두어로 넣어줌. ex) ch(char), str(string), i(int), db(double), b(boolean)

- [ ]  함수가 순수하다는 것은 무엇을 의미하는가

- 숨겨진 정보나 외부 상태가 아닌 결과를 생성하는 입력에만 의존해야 함.
- 참조 또는 전역 변수/ 객체에 의해 전달된 매개변수를 수정하는 것과 같은 관찰 가능한 부작용을 일으키지 않아야 함.

- [ ]  자바스크립트에서 함수를 작성하는 문법은 다음과 같다. 이 외의 함수를 작성하는 문법 두가지가 무엇이며, 어떤 때 사용하나요?

```javascript
    function myFunc(param) {
        statements
}
```

1.  함수 표현식
```javascript
const greet = function(name) {
    return `Hello, ${name}!`;
};

console.log(greet('World')); // 출력: Hello, World!
```
- 익명 함수를 사용할 때
- 콜백 함수로 전달할 때
- 블록 스코프에서 함수가 필요한 경우

2. 화살표 함수
```javascript
const greet = (name) => `Hello, ${name}!`;

console.log(greet('World')); // 출력: Hello, World!
```
- 간결한 문법을 원할 때
- 익명 함수로 사용될 때
- 콜백 함수로 전달할 때
- 'this' 바인딩이 필요 없는 함수

- [ ]  파이썬에서는 문자열을 포맷팅하기 위해 다음과 같은 f-string을 사용합니다. 자바스크립트에서 f-string과 같이 문자열을 다양한 자료형과 변수들을 혼합하여 사용하려면 어떤 방법을 사용하나요?

```javascript
let x = 1;
let y = 2;
let result = `${x} + ${y}는 ${x + y}입니다.`;
console.log(result);  // 출력: 1 + 2는 3입니다.
```

- [ ]  다음과 같은 객체(object)에서 p 값을 출력하려면 o.p 와 같은 방식으로 출력해야 하는데, p 로 출력할 수 있는 방법은 없을까요?

```javascript
var o = { p: 42, q: true };

const { p, q } = o;

console.log(p); // 42
console.log(q); // true
```

**Comment by MrMirror21** [comment]
답변해주신 사전 과제 잘 봤습니다.
전체적으로 잘 답변해주셨고, 오늘 스터디에 미처 담지 못한 순수 함수에 대한 부분도 순수 함수로만 코드를 작성하는게 좋을지 생각해보시면 도움이 될 것 같습니다. 좋은 답변 감사합니다.
